### PR TITLE
Bug/2551 persistent post submission errors

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Adds logic to Clinical Errors
First retrieves Migration Errors, then retrieves Donor records marked with errors. 
If Donor is now Valid, then we filter out any Migration Errors related to this donor.
There will be a follow up ticket (Platform UI #2547) to highlight specific error cells, this PR handles valid/invalid updates and updates general error handling logic

(https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/platform-ui/2551)

**Type of Change**

- [x] Bug
- [x] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/clinical/clinical-service.ts [hunk 1]
```diff
@@ -240,19 +243,84 @@ export const getClinicalEntityMigrationErrors = async (programId: string, query:
...
   }
-
   const end = new Date().getTime() / 1000;
   L.debug(`getClinicalEntityMigrationErrors took ${end - start}s`);
 
-  return clinicalErrors;
+  return { clinicalErrors: clinicalMigrationErrors, migrationLastUpdated };
+};
+
+export const getDonorSubmissionErrorUpdates = async (
```
Comment (joneubank) [inline src/clinical/clinical-service.ts:284]: This function (and also the `getClinicalEntityMigrationErrors`) could benefit from some comments (JSDocs preferable) in order to explain what data these are returning.

File: src/clinical/service-worker-thread/tasks.ts [hunk 2]
```diff
@@ -124,10 +125,13 @@ const sortDocs = (sort: string, entityName: string, completionStats: CompletionR
...
   } else {
+    const currentEntry = currentRecord as ClinicalInfo;
+    const nextEntry = nextRecord as ClinicalInfo;
+
     // Sort by Selected Column
     const key = isDescending === -1 ? sort.split('-')[1] : sort;
-    const first: any = currentRecord[key];
-    const next: any = nextRecord[key];
+    const first: any = currentEntry[key] as string;
+    const next: any = nextEntry[key];
```
Comment (joneubank) [inline src/clinical/service-worker-thread/tasks.ts:134]: do these need to by typed as `any`? would prefer strict types.

**Review by joneubank — APPROVED** [review]
I think we could use more descriptions of the functions in general, but this functionally looks pretty good.

File: src/clinical/service-worker-thread/tasks.ts (hunk 2 above)
Comment (joneubank) [reply src/clinical/service-worker-thread/tasks.ts:134]: just checked this. we should not declare these as any, we know their type from context. They are being checked as `first > next` later which could have weird effects if they are an unexpected type.

Comment (demariadaniel) [reply src/clinical/service-worker-thread/tasks.ts]: I see now. This was definitely a band aid for cascading type issues. I've updated the logic with better types, please take a look!

File: src/clinical/clinical-service.ts (hunk 1 above)
Comment (demariadaniel) [reply src/clinical/clinical-service.ts:298]: OK I've updated comments + added return types to functions

**Comment by ciaranschutte** [comment]
@demariadaniel need to merge latest `develop` changes - CI tests are fixed

**Pull request merged**